src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -649,6 +665,14 @@ public bool TryTake([MaybeNullWhen(false)] out T item, int millisecondsTimeout)
         public bool TryTake([MaybeNullWhen(false)] out T item, int millisecondsTimeout, CancellationToken cancellationToken)
         {
             ValidateMillisecondsTimeout(millisecondsTimeout);
+
+            //If the collection is completed then there is no need remove an item.
+            if (IsCompleted)
+            {
+                item = default(T);
+                return false;
+            }
+
```

Comment:
@pedrobsaila this alters the behavior of the `Take` method in case the collection is completed and the `cancellationToken` is canceled. The existing behavior is to throw an `OperationCanceledException`. The new behavior will be to return `false`. Altering the behavior of the method is unlikely to be desirable. A demonstration of the current behavior can be found in [this](https://github.com/dotnet/runtime/issues/69320#issuecomment-1133587458) comment.

Reply:
I don't think so, `IsCompleted` calls `CheckDisposed`

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L90-L97